// @pothos/plugin-prisma/generated.ts
/* eslint-disable */
import type { Prisma, Student, LearningFocus } from "/Users/meganarnell/Programming Projects/react-graphql-prisma-app/graphql-server/node_modules/@prisma/client";
export default interface PrismaTypes {
    Student: {
        Name: "Student";
        Shape: Student;
        Include: Prisma.StudentInclude;
        Select: Prisma.StudentSelect;
        OrderBy: Prisma.StudentOrderByWithRelationInput;
        WhereUnique: Prisma.StudentWhereUniqueInput;
        Where: Prisma.StudentWhereInput;
        RelationName: "learningFocuses";
        ListRelations: "learningFocuses";
        Relations: {
            learningFocuses: {
                Shape: LearningFocus[];
                Types: PrismaTypes["LearningFocus"];
            };
        };
    };
    LearningFocus: {
        Name: "LearningFocus";
        Shape: LearningFocus;
        Include: Prisma.LearningFocusInclude;
        Select: Prisma.LearningFocusSelect;
        OrderBy: Prisma.LearningFocusOrderByWithRelationInput;
        WhereUnique: Prisma.LearningFocusWhereUniqueInput;
        Where: Prisma.LearningFocusWhereInput;
        RelationName: "student";
        ListRelations: never;
        Relations: {
            student: {
                Shape: Student;
                Types: PrismaTypes["Student"];
            };
        };
    };
}
