// @pothos/plugin-prisma/generated.ts
/* eslint-disable */
import type { Prisma, Student, LearningFocus, WeeklyPlan, Book, DailySchedule, DailyScheduleItem } from "/Users/meganarnell/Programming Projects/react-graphql-prisma-app/graphql-server/node_modules/@prisma/client";
export default interface PrismaTypes {
    Student: {
        Name: "Student";
        Shape: Student;
        Include: Prisma.StudentInclude;
        Select: Prisma.StudentSelect;
        OrderBy: Prisma.StudentOrderByWithRelationInput;
        WhereUnique: Prisma.StudentWhereUniqueInput;
        Where: Prisma.StudentWhereInput;
        RelationName: "learningFocuses";
        ListRelations: "learningFocuses";
        Relations: {
            learningFocuses: {
                Shape: LearningFocus[];
                Types: PrismaTypes["LearningFocus"];
            };
        };
    };
    LearningFocus: {
        Name: "LearningFocus";
        Shape: LearningFocus;
        Include: Prisma.LearningFocusInclude;
        Select: Prisma.LearningFocusSelect;
        OrderBy: Prisma.LearningFocusOrderByWithRelationInput;
        WhereUnique: Prisma.LearningFocusWhereUniqueInput;
        Where: Prisma.LearningFocusWhereInput;
        RelationName: "student";
        ListRelations: never;
        Relations: {
            student: {
                Shape: Student;
                Types: PrismaTypes["Student"];
            };
        };
    };
    WeeklyPlan: {
        Name: "WeeklyPlan";
        Shape: WeeklyPlan;
        Include: Prisma.WeeklyPlanInclude;
        Select: Prisma.WeeklyPlanSelect;
        OrderBy: Prisma.WeeklyPlanOrderByWithRelationInput;
        WhereUnique: Prisma.WeeklyPlanWhereUniqueInput;
        Where: Prisma.WeeklyPlanWhereInput;
        RelationName: "books" | "dailySchedules";
        ListRelations: "books" | "dailySchedules";
        Relations: {
            books: {
                Shape: Book[];
                Types: PrismaTypes["Book"];
            };
            dailySchedules: {
                Shape: DailySchedule[];
                Types: PrismaTypes["DailySchedule"];
            };
        };
    };
    Book: {
        Name: "Book";
        Shape: Book;
        Include: Prisma.BookInclude;
        Select: Prisma.BookSelect;
        OrderBy: Prisma.BookOrderByWithRelationInput;
        WhereUnique: Prisma.BookWhereUniqueInput;
        Where: Prisma.BookWhereInput;
        RelationName: "weeklyPlan";
        ListRelations: never;
        Relations: {
            weeklyPlan: {
                Shape: WeeklyPlan;
                Types: PrismaTypes["WeeklyPlan"];
            };
        };
    };
    DailySchedule: {
        Name: "DailySchedule";
        Shape: DailySchedule;
        Include: Prisma.DailyScheduleInclude;
        Select: Prisma.DailyScheduleSelect;
        OrderBy: Prisma.DailyScheduleOrderByWithRelationInput;
        WhereUnique: Prisma.DailyScheduleWhereUniqueInput;
        Where: Prisma.DailyScheduleWhereInput;
        RelationName: "dailyScheduleItems" | "weeklyPlan";
        ListRelations: "dailyScheduleItems";
        Relations: {
            dailyScheduleItems: {
                Shape: DailyScheduleItem[];
                Types: PrismaTypes["DailyScheduleItem"];
            };
            weeklyPlan: {
                Shape: WeeklyPlan;
                Types: PrismaTypes["WeeklyPlan"];
            };
        };
    };
    DailyScheduleItem: {
        Name: "DailyScheduleItem";
        Shape: DailyScheduleItem;
        Include: Prisma.DailyScheduleItemInclude;
        Select: Prisma.DailyScheduleItemSelect;
        OrderBy: Prisma.DailyScheduleItemOrderByWithRelationInput;
        WhereUnique: Prisma.DailyScheduleItemWhereUniqueInput;
        Where: Prisma.DailyScheduleItemWhereInput;
        RelationName: "dailySchedule";
        ListRelations: never;
        Relations: {
            dailySchedule: {
                Shape: DailySchedule;
                Types: PrismaTypes["DailySchedule"];
            };
        };
    };
}
