// @pothos/plugin-prisma/lib/connection-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    prismaModelKey: ()=>prismaModelKey,
    prismaConnectionHelpers: ()=>prismaConnectionHelpers
});
const _cursors = require("./util/cursors");
const _datamodel = require("./util/datamodel");
const prismaModelKey = Symbol.for('Pothos.prismaModelKey');
function prismaConnectionHelpers(builder, refOrType, options) {
    const modelName = typeof refOrType === 'string' ? refOrType : refOrType.modelName;
    const formatCursor = (0, _cursors.getCursorFormatter)(modelName, builder, options.cursor);
    const parseCursor = (0, _cursors.getCursorParser)(modelName, builder, options.cursor);
    function resolve(list, args, ctx) {
        return (0, _cursors.wrapConnectionResult)(list, args, getQuery(args, ctx).take, formatCursor);
    }
    function getQuery(args, ctx) {
        return (0, _cursors.prismaCursorConnectionQuery)({
            args,
            ctx,
            maxSize: typeof options.maxSize === 'function' ? options.maxSize(args, ctx) : options.maxSize,
            defaultSize: typeof options.defaultSize === 'function' ? options.defaultSize(args, ctx) : options.defaultSize,
            parseCursor
        });
    }
    return {
        ref: typeof refOrType === 'string' ? (0, _datamodel.getRefFromModel)(refOrType, builder) : refOrType,
        resolve,
        getQuery
    };
}
